### PR TITLE
Vendor currency

### PR DESCRIPTION
**Pull request by Forbiddenz**
Introducing Vendor and Currency
- There are bugs in inventory in regards to moving things around packs.
- I will be doing another PR spec to fix up inventory tracking but I did not feel like it was for this PR

### CODE REVIEW

File: Source/ACE/Entity/Container.cs [hunk 1]
```diff
@@ -237,6 +237,32 @@ public virtual WorldObject GetInventoryItem(ObjectGuid objectGuid)
         }
 
         /// <summary>
+        /// This method is used to get all inventory items of Coin in this container (example of usage get all items of coin on player)
+        /// </summary>
+        /// <param name="objectGuid"></param>
```
Comment (Mogwai-TheFurry) [inline Source/ACE/Entity/Container.cs]: if you're not going to fill in param and returns, remove them.

File: Source/ACE/Entity/Player.cs [hunk 2]
```diff
@@ -1196,23 +1191,144 @@ public void HandleActionSellFinalTransaction(WorldObject vendor, bool valid, Lis
...
+                // pay player in voinds
+                if (valid)
+                    CreateCurrancy(WeenieType.Coin, payout);
             }).EnqueueChain();
         }
-#endregion
+        #endregion
 
-        public void HandleAddToInventory(WorldObject wo)
+        public bool CreateCurrancy(WeenieType type, uint amount)
```
Comment (Mogwai-TheFurry) [inline Source/ACE/Entity/Player.cs]: CreateCurrency (spelling)

File: Source/ACE/Entity/Player.cs [hunk 3]
```diff
@@ -1196,23 +1191,144 @@ public void HandleActionSellFinalTransaction(WorldObject vendor, bool valid, Lis
...
         {
             new ActionChain(this, () =>
             {
-                AddToInventory(wo);
-                TrackObject(wo);
+                // todo: we need to look up this object to understand it by its weenie id.
+                // todo: support more then hard coded coin.
+                WorldObject wochk = WorldObjectFactory.CreateNewWorldObject(273);
```
Comment (Mogwai-TheFurry) [inline Source/ACE/Entity/Player.cs]: what is 273?  Put it in a constant and give it a name, please.

File: Source/ACE/Entity/Container.cs (hunk 1 above)
Comment (Forbiddenz) [reply Source/ACE/Entity/Container.cs]: This should be fixed

**Pull request merged**